### PR TITLE
Improve searchwithin labeling structure

### PR DESCRIPTION
**Pull request by devongovett**
Found during code audit. We would like to avoid predictable ids so that they will not conflict with other elements on the page. So this implements an alternative approach to labeling SearchWithin that doesn't require any changes to useSelect.

I am sure the tests are broken. ✌️

### CODE REVIEW

**Review by snowystinger — COMMENTED** [review]
o man, i got so sucked into thinking of the select only inside of a search within I missed that we'd changed pickers too...

great catch, fix makes sense, let me know if you need help fixing the tests

File: packages/@react-spectrum/searchwithin/src/SearchWithin.tsx [hunk 1]
```diff
@@ -127,6 +128,7 @@ function SearchWithin(props: SpectrumSearchWithinProps, ref: FocusableRef<HTMLEl
         role="group"
         className={classNames(styles, 'spectrum-SearchWithin', styleProps.className)}
         ref={groupRef}>
+        <VisuallyHidden id={visuallyHiddenId} data-testid="searchwith-label">{formatMessage('searchWithin')}</VisuallyHidden>
```
Comment (devongovett) [inline packages/@react-spectrum/searchwithin/src/SearchWithin.tsx:131]: btw, I don't think `data-testid` is removed anymore in published builds. maybe we can avoid it and change the test instead?

Comment (snowystinger) [reply packages/@react-spectrum/searchwithin/src/SearchWithin.tsx:131]: Wait, when did that stop?

Comment (devongovett) [reply packages/@react-spectrum/searchwithin/src/SearchWithin.tsx:131]: When we stopped using babel

**Review by dannify — APPROVED** [review]

**Review by reidbarber — APPROVED** [review]
LGTM. Tested VO and NVDA

**Review by ktabors — APPROVED** [review]
This works for me on ipad and iphone. I heard search within read on both the input and picker. On desktop I was able to navigate to more things like the "group" and the VisuallyHidden, which I wasn't on the iPad.

**Pull request merged**